cpp/include/raft/random/rng.cuh

```diff
@@ -291,7 +249,8 @@ class Rng {
    */
   template <typename Type, typename LenType = int>
   void fill(Type *ptr, LenType len, Type val, cudaStream_t stream) {
```

Comment:
It looks like there's quite abit of implementation details for the `Rng` type, including CUDA details. Also, if the hidden impls of the random generator kernels need access to the `Rng` class then the details API is going to be including this file and vice versa. One thing we can do is rename this to something like `RngImpl` and move it to `detail` and then expose it to the public API under the `Rng` name.